plugins/nv-filters/nvidia-audiofx-filter.c

```diff
@@ -259,7 +278,13 @@ bool load_nvidia_afx(void)
 	LOAD_SYM(cuCtxPopCurrent);
 	LOAD_SYM(cuInit);
 #undef LOAD_SYM
-
+#define LOAD_SYM(sym) LOAD_SYM_FROM_LIB2(sym, nv_audiofx, "NVAudioEffects.dll")
+	LOAD_SYM(NvAFX_InitializeLogger);
+	LOAD_SYM(NvAFX_UninitializeLogger);
+	if (!nvafx_new_sdk) {
+		blog(LOG_INFO, "[NVIDIA AUDIO FX]: sdk loaded but old redistributable detected; please upgrade.");
+	}
+#undef LOAD_SYM
```

Comment:
If `NvAFX_InitializeLogger` is not found, but `NvAFX_UninitializeLogger` somehow is, `nvafx_new_sdk` will be set to `true` and the info log message will never be printed.

So as implemented this code has a code path in which it will not do what I think was intended here, particularly as the error checking condition below will also not be used.

Reply:
the two functions are  a balanced pair so it is doubtful one would be available in nvidia sdk redist but not the other. Anyway I've updated the code to protect against a botched redist on nvidia side.

Reply:
They might be balanced internally but you cannot pull them both out simultaneously, but sequentially and as such you either have to check their condition sequentially or check for both of them in a single conditional check.

As the ISO C++ guidelines like to point out:

> Write code that is guaranteed to work, not code that doesn’t seem to break

Thanks for fixing this.